docs/sdk/examples/assets/pastefromoffice/pastefromoffice.js

```diff
@@ -0,0 +1,164 @@
+var sources = [
+    {
+        name: 'Google Docs',
+        link: 'pastefromgoogledocs.html',
+        linkName: 'Paste from Google Docs',
+        regex: /id=(\"|\')?docs\-internal\-guid\-/,
+        isAlreadyShow: false
+    },
+    {
+        name: 'Microsoft Word',
+        link: 'pastefromword.html',
+        linkName: 'Paste from Word',
+        regex: getProperWordRegex(),
+        isAlreadyShow: false
+    },
+    {
+        name: 'Microsoft Excel',
+        link: 'pastefromexcel.html',
+        linkName: 'Paste from Excel',
+        regex: /<meta\s*name=generator\s*content="Microsoft\s*Excel\s*\d+"?\/?/gmi,
+        isAlreadyShow: false
+    },
+    {
+        name: 'LibreOffice',
+        link: 'pastefromlibreoffice.html',
+        linkName: 'Paste from LibreOffice',
+        regex: /<meta\s+name=["']?generator["']?\s+content="LibreOffice\s*\d+"?\/?/gmi,
+        isAlreadyShow: false
+    }
+];
+var instances = getInstances();
+// A state variable that indicates if the clipboard notification has been seen.
+// We use the variable for displaying the notification only once per demo.
+let hasNotificationBeenSeen = false;
+
+instances.forEach( function ( instanceName ) {
+    var editor = CKEDITOR.instances[ instanceName ];
+
+    if ( !editor ){
+        return;
+    }
+
+    editor.on( 'instanceReady', function() {
+        editor.on( 'paste', function( evt ) {
+            var pasteData = CKEDITOR.plugins.pastetools.getClipboardData( evt.data, 'text/html' );
+
+            if( !pasteData ){
+                return;
+            }
+
+            var sourceIndex = getPastedSourceIndex( pasteData );
+            // Show notification only once per pasted source format ( Word, Excel ).
+            if( sourceIndex !== -1 && !hasNotificationBeenSeen && !sources[ sourceIndex ].isAlreadyShow ) {
```

Comment:
Not sure, do we need `hasNotificationBeenSeen` if we are updating `isAlreadyShow` property for every format?